bundle/parallel.go

```diff
@@ -0,0 +1,37 @@
+package bundle
+
+import (
+	"context"
+	"sync"
+
+	"github.com/databricks/cli/libs/diag"
+)
+
+type parallel struct {
+	mutators []Mutator
+}
+
+func (m *parallel) Name() string {
+	return "parallel"
+}
+
+func (m *parallel) Apply(ctx context.Context, b *Bundle) diag.Diagnostics {
+	var wg sync.WaitGroup
+	diags := diag.Diagnostics{}
+	wg.Add(len(m.mutators))
+	for _, mutator := range m.mutators {
+		go func(mutator Mutator) {
+			defer wg.Done()
+			diags = diags.Extend(mutator.Apply(ctx, b))
+		}(mutator)
+	}
+	wg.Wait()
+	return diags
+}
```

Comment:
We can't guarantee that none of the mutators actually mutate something. Could we enforce this somehow?